modeshape-jcr/src/main/java/org/modeshape/jcr/cache/document/DocumentTranslator.java

```diff
@@ -868,15 +868,22 @@ protected long insertChildren( EditableDocument document,
 
     public ChildReferences getChildReferences( WorkspaceCache cache,
                                                Document document ) {
+        Name primaryType = getPrimaryType(document);
+        NodeTypes nodeTypes = getNodeTypes(cache);
+
+        boolean isUnorderedCollection = nodeTypes != null && nodeTypes.isUnorderedCollection(primaryType);
         boolean hasChildren = document.containsField(CHILDREN);
         boolean hasFederatedSegments = document.containsField(FEDERATED_SEGMENTS);
-        if (!hasChildren && !hasFederatedSegments) {
+        if (!hasChildren && !hasFederatedSegments && !isUnorderedCollection) {
             return ImmutableChildReferences.EMPTY_CHILD_REFERENCES;
         }
+
+        if (isUnorderedCollection) {
+            // the base 'mode:unordered' node types do not contain a PRB field...
+            return document.containsField(PRB) ?  new BucketedChildReferences(document, this) : ImmutableChildReferences.EMPTY_CHILD_REFERENCES; 
```

Comment:
Okay, can you add a comment that explains this? The code is not clear without it, and I never would have surmised this is the reason why it might return an `EmptyChildReferences`.